_docs/concepts/encryption.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: guide
+title: StorageOS Docs - Encryption
+anchor: concepts
+module: concepts/encryption
+---
+
+# Encryption
+
+Encryption is enabled on a per volume basis using the
+`storageos.com/encryption` label (for more information see [Encrypted
+Volumes](/docs/operations/encrypted-volumes)).
+
+## Encryption Keys
+
+StorageOS encrypts volumes on disk using AES-256 in XTS-AES mode with 512 bit
+keys as recomended by [NIST](
+https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38e.pdf
+), with encryption keys being derived using [HKDF](
+https://eprint.iacr.org/2010/264.pdf ).
+
+Keys and Initialisation vectors are generated using the [crypto/rand](
+https://godoc.org/crypto/rand ) package.
+
+| Key                   | Size                 | Usage                                                     |
+| :-------------------- | :------------------- | :-------------------------------------------------------- |
+| Volume                | 512 bits random data | Used by XTS-AES to encrypt and decrypt disk blocks        |
+| User                  | 256 bits random data | Used to encrypt Volume keys                               |
+| Initialisation Vector | 256 bits random data | Used as 'salt' when encrypting volume keys with user keys |
+
+Encryption keys are created per volume. However user keys are created per
```

Comment:
I suggest we reword this slightly - it's not totally clear to my eye.

- we create a master key to encrypt a volume
- we then create a per-namespace key and per-volume IV
- the combination of namespace key and IV is used to encrypt master key
- the lot get stored in K8S secrets